### PR TITLE
Seisan sfiles for event

### PR DESCRIPTION
**Pull request by krischer**
This not done by far but I worked a bit with @jsvelasquez to add read support for SEISAN event sfiles to ObsPy.

### CODE REVIEW

File: obspy/io/seisan/sfile.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        is_buffer = False
+
+    if is_buffer is True:
+        ret_val = function(filename_or_buf, *args, **kwargs)
+        if reset_fp:
+            filename_or_buf.seek(position, 0)
+        return ret_val
+    else:
+        with open(filename_or_buf, file_mode) as fh:
+            return function(fh, *args, **kwargs)
```
Comment (megies) [inline obspy/io/seisan/sfile.py]: Thought I've seen this one before.. actually this is the fourth appearance now. Can you do me a favor and move it to core/util? ;-)

Comment (QuLogic) [inline obspy/io/seisan/sfile.py]: Something that @krischer was supposed to do before...

**Comment by krischer** [comment]
I was just made aware that https://github.com/calum-chamberlain/EQcorrscan contains functionality to read seisan catalog files to the ObsPy objects.

Its seems pretty comprehensive and also has tests:

* https://github.com/calum-chamberlain/EQcorrscan/blob/master/eqcorrscan/utils/sfile_util.py
* https://github.com/calum-chamberlain/EQcorrscan/blob/master/eqcorrscan/tests/sfile_util_test.py

@calum-chamberlain: Can we include that in ObsPy? You (and whoever else had a hand in the code) would have to agree that we can relicense it under the LGPLv3. I think that you intend to have this license in any case (at least your root level license file is LGPL) but the license header in the files seem to indicate that its GPL. I'm not sure which one is applicable now.

There is also other stuff that might be useful to have in ObsPy:

* https://github.com/calum-chamberlain/EQcorrscan/blob/master/eqcorrscan/utils/sac_util.py

Is there anything I'm missing? We are happy to take any file formats and generically useful processing things!

If you want to include it into ObsPy yourself: That would of course be the best solution :-)

**Comment by calum-chamberlain** [comment]
@krischer The S-file functionality in EQcorrscan is still pretty basic - it doesn't read in all the header information in the S-files, and I continue to find (and fix) bugs...  I would definitely like it to be in ObsPy - It has just been me a @cjhopp working on these things.  Licence wise, I did mean for it to be under LGPL 3 - I have been meaning to go through all the code and sort the headers.

The sac_util stuff is very early days, that now has a test too.  If you want to take them and develop more you are welcome, but otherwise I was going to keep improving them.

When were you thinking you would want to pull them in? Any idea when 1.1.0 might be out? I can try and develop some more functionality into them before then and pull them in?  I don't think there is anything else really generic in there - would be good to keep EQcorrscan more about the detection stuff!

**Comment by krischer** [comment]
> When were you thinking you would want to pull them in? Any idea when 1.1.0 might be out? I can try and develop some more functionality into them before then and pull them in?

1.1.0 will still take at least a couple of months so there is a lot of time left. Great if you want to add more functionality before then :-) Just let us know once its ready or directly open a pull request to ObsPy! In any case: I think we should shut down this PR as using the s-file features of EQcorrscan seems to be the more promising route. I'll leave this PR open until a new one with the s-file functionality exists as a reminder to us.


> I don't think there is anything else really generic in there - would be good to keep EQcorrscan more about the detection stuff!

Definitely the right approach that most of us also follow: Generic stuff to ObsPy; specialized things to specialized packages.


> The sac_util stuff is very early days, that now has a test too. If you want to take them and develop more you are welcome, but otherwise I was going to keep improving them.

I'll also wait until you deem it ready but I definitely think it might be useful to have in ObsPy.

**Comment by calum-chamberlain** [comment]
Golden, I'll keep you posted.

**Comment by calum-chamberlain** [comment]
Hi all, I have made a new branch and pull request here: https://github.com/obspy/obspy/pull/1517
shall we close this one now?

**Comment by krischer** [comment]
> Hi all, I have made a new branch and pull request here: #1517
> shall we close this one now?

Yes.

**Pull request closed without merging**